examples/mistral/mistral-dpo.yml

```
@@ -0,0 +1,103 @@
+#Note that we are switching from the regular chat template to chatml.
+#If you experience problems with the special tokens, training for more epochs can help.
+#After trainig, merge the model before inference otherwise you might
+#face problems with the special tokens.
+
+base_model: mistralai/Mistral-7B-Instruct-v0.2
+model_type: MistralForCausalLM
+tokenizer_type: LlamaTokenizer
+
+load_in_8bit: false
+load_in_4bit: true
+strict: false
+
+chat_template: chatml
+rl: dpo
+datasets:
+  - path: olivermolenschot/alpaca_messages_dpo_test
+    type: chat_template.default
+    field_messages: conversation
+    field_chosen: chosen
+    field_rejected: rejected
+    message_field_role: role
+    message_field_content: content
+    roles:
+      system:
+        - system
+      user:
+        - user
+      assistant:
+        - assistant
```

```suggestion
    roles:
      system:
        - system
      user:
        - user
      assistant:
        - assistant
```

I don't think this is necessary here. If it doesn't work without this, please let me know!

---

Do you mean to remove the roles? Because your suggestion both removed and added the same thing. I am guessing it is a typo.

---

I removed the roles and the model still trains. So I made the changes in the file and committed.

---

Sorry for that incorrect suggestion. Yes, I meant to remove that. Thank you for doing so.